src/service/storage_arango.py

```diff
@@ -210,4 +224,43 @@ async def get_collection_version_by_num(self, collection_id: str, ver_num: int
             raise errors.NoSuchCollectionVersionError(
                 f"No collection {collection_id} with version number {ver_num}")
         doc = await cur.next()
-        return models.SavedCollection.construct(**_remove_arango_keys(doc))
+        return _doc_to_saved_coll(doc)
+    
+    async def get_collection_versions(
+        self, collection_id: str, max_ver: int = None, limit: int = 1000
+    ) -> list[models.SavedCollection]:
+        """
+        List versions of a collection, sorted from highest to lowest.
+
+        collection_id - the ID of the collection.
+        max_ver - the maximum version to return. This can be used to page through the results.
+        limit - the maximum number of versions to return. Default and maximum value is 1000.
+        """
+        if not limit or limit < 1:
+            limit = 1000
+        if limit > 1000:
+            raise errors.IllegalParameterError("Limit must be <= 1000")
+        if not max_ver or max_ver < 1:
+            max_ver = None
+        bind_vars = {
+            f"@{_FLD_COLLECTION}": _COLL_VERSIONS,
+            _FLD_COLLECTION_ID: collection_id,
+            _FLD_LIMIT: limit
+        }
+        aql = f"""
+                FOR d IN @@{_FLD_COLLECTION}
+                    FILTER d.{models.FIELD_COLLECTION_ID} == @{_FLD_COLLECTION_ID}
+              """
+        if max_ver:
+            bind_vars[_FLD_VER_NUM] = max_ver
+            aql += f"""
+                    FILTER d.{models.FIELD_VER_NUM} <= @{_FLD_VER_NUM}
+                    """
+        aql += f"""
+                    SORT d.{models.FIELD_VER_NUM} DESC
+                    LIMIT @{_FLD_LIMIT}
+                    RETURN d
+                """
+        cur = await self._db.aql.execute(aql, bind_vars=bind_vars)
```

Comment:
is the query tested locally?

Reply:
I tested it via running the server via `docker-compose up`, and also modified the code to hardcode a limit and tested again. Is that what you were looking for?

Reply:
okay. I think that's good for now. I just didn't test it locally.  In case it errors.